From: Michael Grzeschik <m.grzeschik@pengutronix.de>
Date: Wed, 10 Oct 2018 12:36:15 +0200
Subject: [PATCH] dsa: ksz: add test_cnt option for ethtool test options

This patch adds the support to return the test_cnt when running
selft tests on the interface.

Signed-off-by: Michael Grzeschik <m.grzeschik@pengutronix.de>
---
 drivers/net/dsa/microchip/ksz_common.c | 10 +++++++---
 drivers/net/dsa/microchip/ksz_common.h |  2 ++
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/drivers/net/dsa/microchip/ksz_common.c b/drivers/net/dsa/microchip/ksz_common.c
index 7fabc0e3d807..acae01539483 100644
--- a/drivers/net/dsa/microchip/ksz_common.c
+++ b/drivers/net/dsa/microchip/ksz_common.c
@@ -168,10 +168,14 @@ int ksz_sset_count(struct dsa_switch *ds, int port, int sset)
 {
 	struct ksz_device *dev = ds->priv;
 
-	if (sset != ETH_SS_STATS)
-		return 0;
+	switch (sset) {
+	case ETH_SS_STATS:
+		return dev->mib_cnt;
+	case ETH_SS_TEST:
+		return dev->test_cnt;
+	}
 
-	return dev->mib_cnt;
+	return 0;
 }
 EXPORT_SYMBOL_GPL(ksz_sset_count);
 
diff --git a/drivers/net/dsa/microchip/ksz_common.h b/drivers/net/dsa/microchip/ksz_common.h
index 6fc4c4b8c07c..79abc03b000d 100644
--- a/drivers/net/dsa/microchip/ksz_common.h
+++ b/drivers/net/dsa/microchip/ksz_common.h
@@ -77,6 +77,8 @@ struct ksz_device {
 	bool phy_errata_9477;
 	bool synclko_125;
 
+	int test_cnt;
+
 	struct vlan_table *vlan_cache;
 
 	struct ksz_port *ports;
